backend/src/data_entry/mod.rs

```diff
@@ -377,27 +398,37 @@ mod tests {
         assert_eq!(data.voters_counts.poll_card_count, poll_card_count);
 
         // Finalise data entry after correcting the error
-        let response = save(pool.clone(), request_body.clone()).await;
+        let response = save(pool.clone(), request_body.clone(), 1).await;
         assert_eq!(response.status(), StatusCode::OK);
-        let response = finalise(pool.clone()).await;
+        let response = finalise_entry(pool.clone(), 1).await;
         assert_eq!(response.status(), StatusCode::OK);
 
-        // Check if the data entry was finalised:
-        // removed from polling_station_data_entries...
-        let row_count = query!("SELECT COUNT(*) AS count FROM polling_station_data_entries")
+        // Check if the first data entry was finalised:
+        // and that a second entry was created
+        // TODO: We should test the "finalised_at" field and entry numbers
+        let row = query!("SELECT * FROM polling_station_data_entries")
             .fetch_one(&pool)
             .await
             .unwrap();
-        assert_eq!(row_count.count, 0);
-        // ...and added to polling_station_results
+        assert!(row.finalised_at.is_some());
+        // ...and is _not_ added to polling_station_results
```

Comment:
In [babb9f2](https://github.com/kiesraad/abacus/pull/612/commits/babb9f21e4af4f4df9a513223a12922115a31524), I've added a second row count check. The first is to test that nothing yet is inserted into the results table. Since that is how it used to work, it seems like a nice test to keep. The second tests that there is at least something added.